cpp/bench/sg/dbscan.cu

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuML.hpp>
+#include <dbscan/dbscan.hpp>
+#include <utility>
+#include "benchmark.cuh"
+
+namespace ML {
+namespace Bench {
+namespace dbscan {
+
+struct AlgoParams {
+  int min_pts;
+  double eps;
```

Comment:
Should we parameterize this so that we can benchmark the paths for both float and double?

Reply:
for now, I took the shortcut of just keeping it in double and doing a type-cast at the place where we are passing this param to the main algo. I think that's simpler. But I'm open to discussion on this one.

Reply:
Not a huge deal. If we find we need something like that in the future, we can always add it.

Reply:
Thanks.